x-pack/plugins/observability_solution/observability_ai_assistant/common/utils/emit_with_concatenated_message.ts

```diff
@@ -16,8 +16,32 @@ import {
   ConcatenatedMessage,
 } from './concatenate_chat_completion_chunks';
 
+type ConcatenateMessageCallback = (
+  concatenatedMessage: ConcatenatedMessage
+) => Promise<ConcatenatedMessage>;
+
+function mergeWithEditedMessage(
```

Comment:
unrelated change, I'm not sure if mergeMap is supposed to take an async callback, and what happens when the promise is rejected, so I've opted to make it more explicit by using RxJS's `from`, that converts a promise (amongst other things) into an Observable.